tendrl/commons/message.py

```diff
@@ -0,0 +1,87 @@
+import datetime
+from inspect import getframeinfo
+from inspect import stack
+import json
+import uuid
+
+
+class Message(object):
+    # priority[0] is debug....priority[5] is critical
+    # Use priorities and publishers when call create message obj
+    priorities = type("priorities", (),
+                      {"DEBUG": "debug", "INFO": "info", "NOTICE": "notice",
+                       "WARNING": "warning", "ERROR": "error",
+                       "CRITICAL": "critical"})
+    publishers = type("publishers", (),
+                      {"CEPH_INTEGRATION": "ceph-integration",
+                       "GLUSTER_INTEGRATION": "gluster-integration",
+                       "NODE_AGENT": "node-agent",
+                       "COMMONS": "commons",
+                       "ALERTING": "alerting",
+                       "PERFORMANCE_MONITORING": "performance-monitoring"})
+    """When message object intialize, message_id, timestamp and
+
+    caller are None by default, when from_json function call
+    message old message_id, time_stamp and caller is populated
+    """
+    def __init__(self, priority, publisher, payload,
+                 message_id=None, timestamp=None, node_id=None,
+                 caller=None):
+        if message_id is None:
+            self.message_id = str(uuid.uuid4())
+            self.timestamp = datetime.datetime.now().isoformat()
+            # From which function, line and file error raised
+            caller = getframeinfo(stack()[1][0])
+            self.caller = {"filename": (caller.filename.rsplit('/', 1))[1],
```

Comment:
@r0h4n it will tacking caller details from stack, How it will work means it will take last call of this function only. so it will give exactly who called this function